src/server/system_services/bucket_server.js

```diff
@@ -972,19 +972,65 @@ async function delete_bucket_lifecycle(req) {
  * LIST_BUCKETS
  *
  */
+
 async function list_buckets(req) {
-    const buckets_by_name = _.filter(
-        req.system.buckets_by_name,
-        async bucket => await req.has_s3_bucket_permission(bucket, "s3:ListBucket", req) && !bucket.deleting
-    );
-    return {
-        buckets: _.map(buckets_by_name, function(bucket) {
-            return {
-                name: bucket.name,
-                creation_date: bucket._id.getTimestamp().getTime()
-            };
-        })
-    };
+
+    let next_index = 0;
+    let is_truncated = false;
+
+    let continuation_token = req.rpc_params?.continuation_token;
+    const max_buckets = req.rpc_params?.max_buckets;
+
+    const accessible_bucket_list = (
+        await Promise.all(
+            system_store.data.buckets.map(async bucket => (
+                (await req.has_s3_bucket_permission(bucket, "s3:ListBucket", req) && !bucket.deleting) ? bucket : null
+            ))
+        )
+    ).filter(Boolean);
```

Comment:
Why are you running it with `Promise.all`? it is possible to have a **very large** number of buckets.  This can lead to noobaa-core getting stuck or crash

below suggestion should do it serially (check for correctness)

```suggestion
    const accessible_bucket_list = system_store.data.buckets.filter(
        async bucket => await req.has_s3_bucket_permission(bucket, "s3:ListBucket", req) && !bucket.deleting
    );
```

Reply:
Done.